File: Day 9/2507. Smallest Value After Replacing With Sum of Prime Factors.java
class Solution {
    public int prime(int m){
        int a = 0;
        for (int i=2;m>1;i++){
            while(m%i==0){
                a+=i;
                m/=i;
            }
        }
        return a;
    }
    public int smallestValue(int n) {
        int a= prime(n);
        if (a == n){
            return a;
        }
        else{
            return smallestValue(a);
        }
    }
}

File: Day 009/2507. Smallest Value After Replacing With Sum of Prime Factors.py
class Solution(object):
    def prime(self, m):
        a = 0
        i = 2
        while m > 1:
            while m % i == 0:
                a += i
                m //= i
            i += 1
        return a
    
    def smallestValue(self, n):
        a = self.prime(n)
        if a == n:
            return a
        else:
            return self.smallestValue(a)
    
        """
        :type n: int
        :rtype: int
        """